src/main/java/sana/Memory.java

```diff
@@ -0,0 +1,103 @@
+package sana;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.Scanner;

```

Comment:
——— For PR review ———
Nice and clear use of imports, even when `import java.io.*` is tempting to write in projects that extensively use io functionality.